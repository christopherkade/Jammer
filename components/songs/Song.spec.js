import { shallowMount } from '@vue/test-utils'
import cmp from './Song.vue'

describe('[UNIT] Testing the Song component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(cmp, {
      propsData: {
        song: {
          name: 'mockedName',
          artist: 'mockedArtist'
        }
      }
    })
  })

  it('Snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component validation', () => {
    it('should render the main (parent) component', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validtion', () => {
    it('should render the song name', () => {
      expect(wrapper.text()).toContain('mockedName')
    })

    it('should render the song\'s artist', () => {
      expect(wrapper.text()).toContain('mockedArtist')
    })
  })
})
