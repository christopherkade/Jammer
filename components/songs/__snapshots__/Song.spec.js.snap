// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] Testing the Song component Snapshot testing 1`] = `
"<div class=\\"song-card\\"><h1 class=\\"song-title\\">
    mockedName
    <h2 class=\\"song-artist\\">
      mockedArtist
    </h2></h1> <!----></div>"
`;
